src/mem/mediumslab.h

```diff
@@ -82,13 +82,13 @@ namespace snmalloc
     }
 
     template<ZeroMem zero_mem, SNMALLOC_CONCEPT(ConceptPAL) PAL>
-    void* alloc(size_t size)
+    static void* alloc(Mediumslab* self, size_t size)
```

Comment:
It would be good to have a comment explaining why this is a `static` method with an explicit `this`.  Same comment on the next ones.